test/object-store/sync/flx_sync.cpp

```diff
@@ -174,9 +173,13 @@ TEST_CASE("flx: connect to FLX-enabled app", "[sync][flx][baas]") {
             subs.get_state_change_notification(sync::SubscriptionSet::State::Complete).get();
         }
 
-        wait_for_download(*realm);
         {
-            wait_for_advance(*realm);
+            timed_wait_for(
```

Comment:
I think `realm->refresh()` does the trick here (same everywhere this is used)

Reply:
`wait_for_advance()` exists specifically to verify that flx correctly triggers autorefreshes, which both `refresh()` and polling fail to do.